tss-esapi/tests/integration_tests/context_tests/general_esys_tr_tests.rs

```diff
@@ -445,3 +447,33 @@ mod test_tr_from_tpm_public {
         assert_eq!(expected, actual);
     }
 }
+
+mod test_tr_serialize_tr_deserialize {
+    use super::*;
+
+    #[test]
+    fn test_tr_serialize_tr_deserialize() {
+        let mut context = create_ctx_with_session();
+        let random_digest = context.get_random(16).unwrap();
+        let key_auth = Auth::from_bytes(random_digest.as_bytes()).unwrap();
+        let key_handle = context
+            .create_primary(
+                Hierarchy::Owner,
+                decryption_key_pub(),
+                Some(key_auth),
+                None,
+                None,
+                None,
+            )
+            .unwrap()
+            .key_handle;
+        let data = context.tr_serialize(key_handle.into()).unwrap();
+        // Load handle in a new context
+        let mut new_context = create_ctx_with_session();
+        let new_handle = new_context.tr_deserialize(&data).unwrap().into();
+        assert_eq!(
```

Comment:
Have you checked if the TPM just returns the same handle to you? (That would be sort of OK in this specific case but perhaps not what you wanted to test).

Reply:
Disregard this I read your initial comment regarding this more closely.

Reply:
This is a little bit dangerous and will not work for all tests. You should probably drop the first context before creating the second otherwise the creation of a context will be blocking forever.

Reply:
*TSS ESAPI v1p0 r14, section 7.1 Esys_TR_Serialize()*
> NOTE: The auth value has to be set again after deserialization of the object.

I think you need to add the auth to the deserialized handle before reading public?